### DISCUSSION TITLE
How to determine VPC an RDS instance is associated to?

### DISCUSSION THREAD

**Discussion by lcmgh** [awslabs/aws-sdk-rust#876] [Q&A] [answered]
It looks to me that `aws_sdk_rds::types::_db_cluster` does not contain that while the info `vpc_security_groups` is available.

**Comment by jmklix** [comment]
You should be able to get this with [VpcSecurityGroupMembership](https://docs.rs/aws-sdk-rds/latest/aws_sdk_rds/types/struct.VpcSecurityGroupMembership.html) (and [DbSubnetGroup](https://docs.rs/aws-sdk-rds/latest/aws_sdk_rds/types/struct.DbSubnetGroup.html) will have even more info)

It would look like this:
```rust
let client = aws_sdk_rds::Client::new(&config);
let result = client.describe_db_instances().send.await?; 
//optionally pass db_instance_identifier or filters (https://docs.rs/aws-sdk-rds/latest/aws_sdk_rds/operation/describe_db_instances/struct.DescribeDbInstancesInput.html)

for db_instance in result.db_instances().unwrap_or_default() {
    println!(
        "DB security groups: {:?}",
        db_instance
            .vpc_security_groups().unwrap_or_default().vpc_security_group_id().unwrap()
            .expect("instance should have a security group")
    );
    println!(
        "DB subnet group:    {:?}",
        db_instance
            .db_subnet_group().unwrap_or_default().db_subnet_group_name().unwrap()
            .expect("instance should a subnet group")
    );
}
```

**Comment by lcmgh** [comment, marked as answer]
I just found that `pub fn db_subnet_group(&self) -> ::std::option::Option<&str>` does not return a `struct DbSubnetGroup` and `VpcSecurityGroupMembership` only has `vpc_security_group_id` and has `status`.

Edit: I may need to use https://docs.rs/aws-sdk-rds/latest/aws_sdk_rds/operation/describe_db_subnet_groups/struct.DescribeDbSubnetGroupsOutput.html#method.db_subnet_groups
